core/forkid/forkid.go

```diff
@@ -203,6 +206,46 @@ func newFilter(config *params.ChainConfig, genesis common.Hash, headfn func() ui
 			// No exact, subset or superset match. We are on differing chains, reject.
 			return ErrLocalIncompatibleOrStale
 		}
+		// Verify forks by time
```

Comment:
This looks extremely similar to the work done above, could we break this out into a separate generic function and run it over the forks by block and by time separately?

Reply:
I tried, but couldn't make it work as you need to compare head vs time in the middle of the function. I can give it another try, was pretty tired when I wrote that code

Reply:
Okay, got it down to only a few lines of pasta